Source/WebCore/Modules/webxr/WebXRSpace.h

```diff
@@ -73,18 +73,21 @@ class WebXRSpace : public EventTarget, public ContextDestructionObserver {
 // https://immersive-web.github.io/webxr/#xrsession-viewer-reference-space
 // This is a helper class to implement the viewer space owned by a WebXRSession.
 // It avoids a circular reference between the session and the reference space.
-class WebXRViewerSpace : public WebXRSpace {
+class WebXRViewerSpace : public RefCounted<WebXRViewerSpace>, public WebXRSpace {
     WTF_MAKE_ISO_ALLOCATED(WebXRViewerSpace);
 public:
     WebXRViewerSpace(Document&, WebXRSession&);
```

Comment:
Refcounted classes should never have a public constructor. That too error prone. Please make the constructor private and add a public factory function instead:
```
static Ref< WebXRViewerSpace> create(Document& document, WebXRSession& session)
{
    return adoptRef(*new WebXRViewerSpace(document, session));
}
```